edgedb/datatypes/object.c

```diff
@@ -365,6 +369,15 @@ EdgeObject_InitType(void)
         return NULL;
     }
 
+    // Pass the `dataclasses.is_dataclass(obj)` check, the dict is always empty
+    PyObject *default_fields = PyDict_New();
+    if (default_fields == NULL) {
+        return NULL;
+    }
+    PyDict_SetItemString(
+        EdgeObject_Type.tp_dict, "__dataclass_fields__", default_fields
```

Comment:
Why is this needed if we have a getter?

Reply:
In order to pass the `dataclasses.is_dataclass(obj)` check, which checks `hasattr(type(obj), "__dataclass_fields__")`

Reply:
> which checks hasattr(type(obj), "__dataclass_fields__")

Can you add a comment about that?

Reply:
Sure - the comment is more complete now